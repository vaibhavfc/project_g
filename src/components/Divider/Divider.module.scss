@import '../../sass/theme/default';

.root {
  background-color: $divider-bg;
}

.smallHorizontal {
  height: 1px;
  width: 99vw;
}

.largeHorizontal {
  height: 0.75rem;
  width: 99vw;
}

.largeVertical {
  width: 0.75rem;
  height: 99vh;
}

.smallVertical {
  width: 1px;
  height: 99vh;
}
