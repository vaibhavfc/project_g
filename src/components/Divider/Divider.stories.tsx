// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Divider from '.';

import { DividerProps } from './Divider.type';

export default {
  title: 'Components/Divider',
  component: Divider,
  parameters: {
    controls: {
      // disabled: true,
    }
  }
} as Meta;

const Template: Story<DividerProps> = (args) => <Divider {...args} />

export const HorizontalSmall = Template.bind({});
HorizontalSmall.args = {
  orientation: 'horizontal',
  size: 'small'
}

export const HorizontalLarge = Template.bind({});
HorizontalLarge.args = {
  orientation: 'horizontal',
  size: 'large'
}

export const VerticalSmall = Template.bind({});
VerticalSmall.args = {
  orientation: 'vertical',
  size: 'small'
}

export const VerticalLarge = Template.bind({});
VerticalLarge.args = {
  orientation: 'vertical',
  size: 'large'
}
