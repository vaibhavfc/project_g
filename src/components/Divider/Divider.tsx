import React, { FC, forwardRef, useMemo } from 'react';
import { Divider as MuiDivider } from '@material-ui/core';
import classnames from 'classnames';
import { DividerProps, DividerStylingProps } from './Divider.type';

import classes from './Divider.module.scss';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const DividerWithInnerRef: FC<DividerProps> = ({
  innerRef = null,
  classes: overrideClasses = {},
  orientation,
  size,
  ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<DividerStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );
  //! WIDTH - need to change
  return (
    <MuiDivider
      orientation={orientation}
      classes={{
        root: classnames(
          mergedClasses.root, {
            [classes.smallHorizontal]: (size === 'small' && orientation === 'horizontal'),
            [classes.largeHorizontal]: (size === 'large' && orientation === 'horizontal'),
            [classes.largeVertical]: (size === 'large' && orientation === 'vertical'),
            [classes.smallVertical]: (size === 'small' && orientation === 'vertical'),
          },
        ),
      }}
    />
  );
};

const Divider = forwardRef<HTMLHRElement, Omit<DividerProps, 'innerRef'>>(
  (props, ref) => <DividerWithInnerRef innerRef={ref} {...props} />,
);
export default Divider;
