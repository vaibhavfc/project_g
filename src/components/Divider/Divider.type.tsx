import { Ref } from 'react'
import { DividerProps as MuiDividerProps, DividerClassKey } from '@material-ui/core/Divider'
export interface DividerStylingProps extends Partial<Record<DividerClassKey, string>> {
}

export interface DividerProps extends MuiDividerProps {
  innerRef: Ref<HTMLHRElement>
  size: 'small' | 'large';
}
