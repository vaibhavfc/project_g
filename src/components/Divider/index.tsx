import Divider from './Divider';
import { DividerProps } from './Divider.type';

export type { DividerProps }

export default Divider;