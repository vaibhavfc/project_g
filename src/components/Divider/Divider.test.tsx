import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Divider from './Divider';

describe('test', () => {
  it('Renders Divider Compoents', () => {
    render(<Divider>test</Divider>);
  });

  it('Matches Snapshot of Divider', () => {
    const tree = renderer.create(<Divider> TEST</Divider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
