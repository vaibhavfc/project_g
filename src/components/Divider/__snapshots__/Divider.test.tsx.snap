// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Divider 1`] = `
<hr
  className="MuiDivider-root"
/>
`;
